### ISSUE TITLE
Каких возможностей не хватает / что реализовано плохо

### ISSUE THREAD

**Issue by Tuch** [Tuch/angular-dnd#13]
Прошу сюда писать пожелания, что бы знать, что улучшать в первую очередь

**Comment by 8026787** [comment]
В примере с лассо - лассо позволяет выделить несколько объектов, но перетаскивать все выделенные - нет.

**Comment by EzheG** [comment]
Здравствуйте. Прекрасный модуль! Обнаружил пару недочетов:
1. При использовании `dnd-draggable-opts="{helper:'clone'}"` у перемещаемого объекта нет смещения отноительно точки где мы его захватили. Позже в пулреквесте предложу свой вариант исправления.
2. На touch девайсах нужен небольшой delay чтобы мы могли скролить страницу/список если попали пальцем в dnd объект.

**Comment by Tuch** [comment]
Хорошая идея, буду рад помощи, т.к. сейчас мало времени заниматься проектом :)




С Уважением, 
Афонин А.В.

**Comment by sanyam-j** [comment]
Как я ограничить размер, чтобы максимально в стороны контейнера ?

**Comment by Tuch** [comment]
you can use something like  dnd-on-resizeend = "ctrl.dragend()" dnd-rect = "ctrl.rect", where ctrl is:

```
function ctrl () {
    this.rect = {top: 100, left: 100, width: 100, height: 100}
    
    var maxWidth = 300;
    this.dragend = function () {
       if (parseInt(this.rect.width) > maxWidth) this.rect.width = maxWidth;
    }
}
```

**Comment by sanyam-j** [comment]
Hi,

I am using your first demo of dragging, re-sizing and rotating. While dragging its working fine by not dragging element outside the container. But while re-sizing, for exp my container is of 500px and my re-sizable element is of 50px originally and is placed in center of the container and also i have given it the maxWidth 500px, so while i re-sizing it to right, it get re-sized to 500px and go out of the container. So what i need is while re-sizing, the element can't be re-sized more than its container.

Secondly, i want to edit the text written in element?

I am stuck because of it for 2 3 days. Please help.

Thanks !!!

**Comment by Tuch** [comment]
This was a bag. You can use new version (0.1.11) where this bag was fixed. Also add new dnd-containment = "parent selector" directive instead dnd-container and restrictTheMovement option. Please look demo sources

**Comment by sanyam-j** [comment]
@Tuch Thanks !!! The only thing left for me is to edit the text using any editor on the fly. While i ma using the textangular, its toolbar is not focusing properly.

**Comment by sanyam-j** [comment]
Rotation & Re-sizing is not working good in touch devices. Do i need to set some variable for it?

**Comment by Tuch** [comment]
Very strange. I just tested demo in http://tuch.github.io/angular-dnd/latest/demo/ link and it works fine..

**Comment by sanyam-j** [comment]
@Tuch I am using it in nexus 5 chrome. Its getting difficult to rotate and re-size. Not an issue, i think screen is small so touching the exact point is getting difficult.

I am trying to add textangular text editor inside the element. Its not working and can you help?

**Comment by Tuch** [comment]
can you show jsfiddle?

**Comment by sanyam-j** [comment]
I have deployed test application at http://intense-wildwood-2005.herokuapp.com/ . And the repo is https://github.com/sanyam-j/test_application. Its not working properly. Can you help me out?

**Comment by Tuch** [comment]
check this example http://plnkr.co/edit/3tYZav1CTffeITYJityr?p=preview. It what you need?

**Comment by sanyam-j** [comment]
@Tuch Yes i want this only. But when i go this with newer version of angular, it throws the following execption https://docs.angularjs.org/error/$compile/multidir?0=dndDraggable&p1=textAngular&p2=new

**Comment by Tuch** [comment]
try this my friend http://plnkr.co/edit/3tYZav1CTffeITYJityr?p=preview
This error meant that more than one directive requires its own scope (scope: true in directive definition)

**Comment by sanyam-j** [comment]
@Tuch Yes i am doing this way only. But the problem with this is that when i am decreasing the size of element, its not decreasing the size of inner textangular div and making it float over it

**Comment by Tuch** [comment]
oh, i mind u must do some magic with css..

**Comment by sanyam-j** [comment]
@Tuch  Hi. I have done lot of things with css and its working. Thanks.
One last issue i am facing is when add dragging to element, its not possible to select the text inside that div. As i found, that it disable the select over that div. Can you help me out on it. it will be great help.

**Comment by Tuch** [comment]
try use this https://github.com/Tuch/angular-dnd#dnd-pointer-none option

**Comment by sanyam-j** [comment]
@Tuch Thanks buddy. Its works for me. I am tried this earlier but forget it. Really Thanks !!

**Comment by sanyam-j** [comment]
@Tuch Hi, when i have some more html element inside the editable div, which i am using as input and have dnd-pointer-none. The dnd-pointer-none doesn't work on inner elements because it is on the parent div and not on the child elements. Can you help me out?

**Comment by Tuch** [comment]
can u show jsfiddle example?

**Comment by sanyam-j** [comment]
@Tuch Here's the example for what i am saying. You can see that my model variable has html in it, so when i place dnd-pointer-none on parent and if it has child elements, then dnd-pointer-none does't work.

**Comment by Tuch** [comment]
@sanyam-j try new 0.1.12 version. I fix this

**Comment by sanyam-j** [comment]
@Tuch Thanks !!

**Comment by Carw** [comment]
Можно ли сделать многоуровневое дерево?

**Comment by Tuch** [comment]
вы имеете ввиду вложенные dnd элементы?
что-то вроде этого http://plnkr.co/edit/lzqN3VmEblAr700M1bUS?p=preview

**Comment by wegorich** [comment]
Не хватает примера или описания css классов

**Comment by mfyodorov** [comment]
В директиве dndSortable когда формируется объект настроек dnd-draggable-opts параметр handle задается как handle: opts.handle, но opts.handle - строка и может быть селектором с точкой вначале, и в итоге получается невалидное значение, когда данный объект записывается в атрибут элемента. Это вызывает ошибку. Нужно поправить на handle: "'"+opts.handle+"'"

**Comment by tarankov88** [comment]
Не хватает мульти D&D, например когда мы через лассо выделяем группу элементов, они получают статус "выделенные", хотелось бы при перетаскивании перемещать сразу все выделенные элементы, как минимум опционально. Благодарю!
